include/aws/common/common.h

```diff
@@ -128,6 +128,9 @@ AWS_STATIC_ASSERT(sizeof(intptr_t) == sizeof(void *));
 AWS_STATIC_ASSERT(sizeof(char) == 1);
 #endif
 
+#define AWS_MIN(a, b) ((a) < (b) ? (a) : (b))
+#define AWS_MAX(a, b) ((a) > (b) ? (a) : (b))
```

Comment:
This may lead results not expected by the user if `a` or `b` have side effects or are `volatile` or otherwise shared variables. Other libraries would use code along the lines of `#define AWS_MIN(a, b) ({__typeof__(a) _a = a; __typeof__(b) _b = b; _a < _b ? _a : _b; })`

Reply:
That'll work for gcc and clang, lemme see what msvc wants for that.